index.js

```diff
@@ -2056,23 +2060,23 @@
       }
       xs.splice(lower, 0, {x: x, fx: fx});
       return xs;
-    }, [], foldable);
+    }, []);
 
     var F = foldable.constructor;
-    var result = empty(F);
+    var result = Monoid.methods.empty(F)();
     for (var idx = 0; idx < rs.length; idx += 1) {
-      result = concat(result, of(F, rs[idx].x));
+      result = Semigroup.methods
+        .concat(result)(Applicative.methods.of(F)(rs[idx].x));
     }
     return result;
   }
 
-  //# takeWhile :: (Applicative f, Foldable f, Monoid (f a)) => (a -> Boolean, f a) -> f a
```

Comment:
I took the liberty of changing the signature and description of both `takeWhile` and `dropWhile`. They've been out of date since the introduction of `Filterable`.

Reply:
Thank you!